tests/functional/setups/manu_sneddon_2d.py

```diff
@@ -0,0 +1,445 @@
+        np.arctan2(xf[1, :], (xf[0, :] - h)) - np.arctan2(xf[1, :], (xf[0, :] + h))
+    )
+
+    # The following equalities can be found on page 91 where f3,f4 as equation (5.5.3) and ux, uy in (5.5.1)
+    # Also this is in the coordinate system described in (4.5.1) at page 57, which essentially is a rotation.
+    f4[:] = m * (
+        xf[1, :] / ((xf[0, :] - h) ** 2 + xf[1, :] ** 2)
+        - xf[1, :] / ((xf[0, :] + h) ** 2 + xf[1, :] ** 2)
+    )
+
+    f5[:] = m * (
+        (xf[0, :] - h) / ((xf[0, :] - h) ** 2 + xf[1, :] ** 2)
+        - (xf[0, :] + h) / ((xf[0, :] + h) ** 2 + xf[1, :] ** 2)
+    )
+
+    u[0, bound_faces] = du * (
+        -(1 - 2 * poi) * np.cos(alpha) * f2[:]
+        - 2 * (1 - poi) * np.sin(alpha) * f3[:]
+        - xf[1, :] * (np.cos(alpha) * f4[:] + np.sin(alpha) * f5[:])
+    )
+    u[1, bound_faces] = du * (
+        -(1 - 2 * poi) * np.sin(alpha) * f2[:]
+        + 2 * (1 - poi) * np.cos(alpha) * f3[:]
+        - xf[1, :] * (np.sin(alpha) * f4[:] - np.cos(alpha) * f5[:])
+    )
+
+    return u
+
+
+def assign_bem(
+    grid: pp.Grid,
+    h: float,
+    bound_faces: np.ndarray,
+    alpha: float,
+    bem_centers: np.ndarray,
+    u_a: np.ndarray,
+    poi: float,
+) -> np.ndarray:
+    """Computes semi-analytical displacement values using the BEM for the Sneddon
+    problem.
+
+    Parameter
+        grid: The matrix grid.
+        h: BEM segment length.
+        bound_faces: Array of indices of boundary faces of ``grid``.
+        alpha: Fracture orientation.
+        bem_centers: BEM segments centers.
+        u_a: Sneddon's analytical relative normal displacement.
+        poi: Poisson ratio.
+
+    Return:
+        Semi-analytical boundary displacement values.
+
+    """
+
+    bc_val = np.zeros((grid.dim, grid.num_faces))
+
+    alpha = np.pi / 2 - alpha
+
+    bound_face_centers = grid.face_centers[:, bound_faces]
+
+    for i in range(0, u_a.size):
+        new_bound_face_centers = transform(bem_centers[:, i], bound_face_centers, alpha)
+
+        u_bound = get_bc_val(
+            grid, bound_faces, new_bound_face_centers, h, poi, alpha, u_a[i]
+        )
+
+        bc_val += u_bound
+
+    return bc_val
+
+
+class ManuSneddonExactSolution2d:
+    """Class representing the analytical solution for the pressurized fracture problem."""
+
+    def __init__(self, params: dict):
+        self.p0 = params.get("p0")
+        self.theta = params.get("theta")
+
+        self.a = params.get("a")
+        self.shear_modulus = params.get("material_constants").get("solid").shear_modulus
+        self.poi = params.get("poi")
+        self.length = params.get("length")
+        self.height = params.get("height")
+
+    def exact_sol_global(self, sd: pp.Grid) -> np.ndarray:
+        """Compute the analytical solution for the pressurized fracture problem in
+        question.
+
+        Parameters:
+            sd: Subdomain for which the analytical solution is to be computed.
+
+        """
+
+        n = 1000
```

Comment:
This `n` is a number of bem segments. Should it be hardcoded here? Will it not increase with grid refinement?

Reply:
You're right, it should not be hardcoded here. Perhaps retrieve from params as e.g. "num_bem_segments"?